Use new data

StefanieJaeger opened the pull request:
I included the fix from #13 and fixed the problem, that navigating to an url with an old module would result in an empty plan.

jeremystucki reviewed (dismissed):
Please wait with merging until 30.04.2023 23:00 Uhr

StefanieJaeger commented:
Following things have been tested:

- [ ] I selected a "new" URL => All modules were correctly parsed and rendered
    - I selected "Teilzeit Software Engineering", which contains the "new" modules FP and BAI21
- [ ] I tried an "old" URL => The old modules were replaced with the newer versions, all modules were correctly parsed and rendered
    -  I copied a "new" URL, edited it to be old (FunProg and BAI14) and removed a module (to be sure, that the URL was correctly parsed and not cached) and pasted it in a new tab
- [ ] I selected different semesters and checked the progress bar for the ECTS
- [ ] I added and removed some modules and checked the progress bar and focuses
- [ ] I selected some different default plans and checked the progress bar and focuses

StefanieJaeger commented:
I had to update dependencies, in order to even get it working locally. So this PR also kind of resolves #11.

Mafii commented on src/views/Home.vue:
```diff
@@ -217,18 +221,21 @@ export default {
-      if (cachedPlan != null) {
-        window.location.hash = cachedPlan;
+        if (newPath !== path) {
+          window.location.hash = newPath.replace('#/', '');
```
Is that all that has to be done?

Mafii replied:
Or what is the #/ replacement for?

StefanieJaeger replied:
I'm not quite sure what you mean with "Is that all that has to be done?".

Regarding the "/#" I'm not quite sure anymore. I just tested it without the replacement and it worked just fine.
I probably assumed that we had to replace it to not get 2 # in the url. But I'll remove the replacement, since that does not seem to be the case.

Mafii replied:
I misunderstood something, but it's fine now!

Mafii commented:
With this, my Italian is suddenly 0 ects - which is wrong imo. It is permitted to have 4 ects totally free of choice no matter the category - what's happening here?

![image](https://github.com/lost-university/web/assets/10061519/ef72ed15-cdea-44cf-a169-652220a6fdc5)


Current live-version:

![image](https://github.com/lost-university/web/assets/10061519/7cb95c13-35e5-4566-a4b2-d033df49f41f)

Mafii commented on src/views/Home.vue:
```diff
@@ -199,15 +191,32 @@ export default {
+
       if (path.startsWith(planIndicator)) {
-        return path
+        // This ensures backwards compatability. Removing it after everyone who started before 2022
```
@StefanieJaeger I've allowed myself to push this commit - I have two reasons for that:

My plan (in my cache) caused an error, so I added SEP1 and SEP2 migrations.

Additionally, I want to be able to check my plans in the next two years without manually starting an old version - and I have them in my saved messages on telegram (so local storage is not sufficient) where I copy them from. They still contain SE1 and 2, so I kind of don't want that support removed right now.

Is that okay?

Mafii replied:
I'm sure I'm not the only one :)

jeremystucki replied:
I thought we already had this somewhere, but I can't find it.

jeremystucki replied:
Btw @Mafii I'd prefer if you could use GitHub's "Suggest Changes" feature instead of pushing directly in the future. This way the PR author keeps the control.

StefanieJaeger replied:
I would have also preferred a code suggestion for me to accept.
But the code is fine like this.

Mafii replied:
sorry, will do next time!

Mafii commented:
Looks good to me otherwise @StefanieJaeger !

StefanieJaeger commented:
> With this, my Italian is suddenly 0 ects - which is wrong imo. It is permitted to have 4 ects totally free of choice no matter the category - what's happening here?
> 
> ![image](https://user-images.githubusercontent.com/10061519/250347640-ef72ed15-cdea-44cf-a169-652220a6fdc5.png)
> 
> Current live-version:
> 
> ![image](https://user-images.githubusercontent.com/10061519/250347651-7cb95c13-35e5-4566-a4b2-d033df49f41f.png)

I fixed this in https://github.com/lost-university/data/pull/3.

StefanieJaeger commented:
I'll wait with merging this PR until after the 2nd phase of  the "Stundenplananmeldung" is over. Just in case.
